### DISCUSSION TITLE
Setting up and configuring Minio and uploading a Parquet data file

### DISCUSSION THREAD

**Discussion by alberttwong** [StarRocks/starrocks#22782] [How-tos, Tutorials, Best Practices and Architecture Articles]
<img referrerpolicy="no-referrer-when-downgrade" src="https://static.scarf.sh/a.png?x-pxid=6153a87f-aa31-4a01-accf-e1ac7244b27c" />

This tutorial describes how you can setup and configure Minio.

# Prerequisites

For this tutorial you need to:
- Have Docker Desktop or podman container runtime installed
- Download a Snowflake provided Parquet data file.
- Download and configure Minio (an open source object store that you can install on your own workstation) and upload the parquet file into the bucket.

# Docker Desktop or podman container runtime installed
This is out of scope for the tutorial.  Please see Docker Desktop or podman's website on how to install that software.

# Downloading the Sample Data File
To download the sample Parquet data file, click [cities.parquet](https://docs.snowflake.com/en/_downloads/0c1e6c4f4140561029eeb20afdd02664/cities.parquet).

The Parquet data file includes sample continent data. The following is a representative example:
```
{
  "continent": "Europe",
  "country": {
    "city": [
      "Paris",
      "Nice",
      "Marseilles",
      "Cannes"
    ],
    "name": "France"
  }
}
```

# Download and configure Minio (an open source object store that you can install on your own workstation) and upload the parquet file into the bucket
The following commands will download Minio, configure a bucket, create credentials and upload the parquet file into the bucket

## Step 1: Run Minio
Going to https://min.io/download#/docker
```
podman run -p 9000:9000 -p 9001:9001 minio/minio server /data --console-address ":9001"
```

## Step 2: Login to Minio
Login the UI at http://localhost:9001
![Screenshot 2023-04-28 at 10 17 05 AM](https://user-images.githubusercontent.com/749093/235212675-1b8431c1-b1b1-4f73-bcee-f87e53bf308a.png)

## Step 3: Create S3 compatible bucket
Click on "creating a new bucket".  I named it "albertwong".

Issue: There is an outstanding github issue with bucket names with "." in them.  Github Issue  https://github.com/StarRocks/starrocks/issues/22711

![Screenshot 2023-04-28 at 10 17 26 AM](https://user-images.githubusercontent.com/749093/235212710-40aaee6b-adf9-4963-a789-d5772c070910.png)
![Screenshot 2023-04-28 at 10 17 39 AM](https://user-images.githubusercontent.com/749093/235212719-4edc285a-a769-4bd4-abf8-63a68da9047f.png)
![Screenshot 2023-04-28 at 10 17 49 AM](https://user-images.githubusercontent.com/749093/235212935-2c94fcd4-e5f7-4c80-b256-9e4bd8dce2bc.png)

## Step 4: Create credentials
Then you need credentials.  Make sure you save the access_key and secret_key since you'll need that in the next steps.
![Screenshot 2023-04-28 at 10 18 02 AM](https://user-images.githubusercontent.com/749093/235212999-8d9d0169-bb31-4cfb-9a37-966f7239ef47.png)
![Screenshot 2023-04-28 at 10 18 14 AM](https://user-images.githubusercontent.com/749093/235213023-bfb9637f-b9fe-4a23-88c1-147c8f605334.png)
![Screenshot 2023-04-28 at 10 18 31 AM](https://user-images.githubusercontent.com/749093/235213078-3048235d-cb41-44f7-8bf5-9c71baae4b16.png)

## Step 5: Upload parquet file
Now you'll upload the parquet file.  To double check, make sure the file as been uploaded.
![Screenshot 2023-04-28 at 10 18 50 AM](https://user-images.githubusercontent.com/749093/235213090-629925ef-e50c-4cd0-a85a-de1cdc02a09b.png)
![Screenshot 2023-04-28 at 10 19 02 AM](https://user-images.githubusercontent.com/749093/235213109-f2621a6d-2c73-46f0-a363-b9270a904874.png)
![Screenshot 2023-04-28 at 10 19 16 AM](https://user-images.githubusercontent.com/749093/235213119-102d18d1-9049-4a61-bb0d-e11f40146bec.png)
![Screenshot 2023-04-28 at 10 19 28 AM](https://user-images.githubusercontent.com/749093/235213130-ac39c76f-380b-49c4-97fa-ab43e9f891e9.png)

**Comment by masalinas** [comment]
Sorry but in this steps, you don't comment anything about configure parquet format in minio because this functionality is disable by default. I tried to add the environment variable **MINIO_API_SELECT_PARQUET** set to **on** in tje operator, but I can't read any parquet file with this error:

`botocore.exceptions.ClientError: An error occurred (InternalError) when calling the SelectObjectContent operation (reached max retries: 4): We encountered an internal error, please try again.: cause(parquet format parsing not enabled on server)`
